services/jetbrains/jetbrains-rating.service.js

```diff
@@ -70,9 +74,26 @@ module.exports = class JetbrainsRating extends JetbrainsBase {
   }
 
   async handle({ format, pluginId }) {
-    const pluginData = await this.fetchPackageData({ pluginId, schema })
-    const pluginRating =
-      pluginData['plugin-repository'].category['idea-plugin'][0].rating
-    return this.constructor.render({ rating: pluginRating, format })
+    let rating
+    if (this.constructor._isLegacyPluginId(pluginId)) {
+      const intelliJPluginData = await this.fetchIntelliJPluginData({
+        pluginId,
+        schema: intelliJschema,
+      })
+      rating =
+        intelliJPluginData['plugin-repository'].category['idea-plugin'][0]
+          .rating
+    } else {
+      const jetbrainsPluginData = await this._requestJson({
+        schema: jetbrainsSchema,
+        url: `https://plugins.jetbrains.com/api/plugins/${this.constructor._cleanPluginId(
+          pluginId
+        )}/rating`,
+        errorMessages: { 400: 'not found' },
+      })
```

Comment:
Would it be worthwhile to pull any of this up into the base class or helper functions? Seems like there will be a lot of repetition for at least some parts (e.g. the json calls)

Reply:
Doesn't matter all that much to me, but seems like the only differences are the value for the schema and an optional/additional trailing path segment on the uri which could both be handled in a single foo/fetch type function (we do something similar for Jenkins with more complex variances on the calls)